File: observer/java/Samsung.java
package Observer;

// Observer sınıfından türer.
// UML diyagramındaki ConcreteObservable yapısına denk gelmektedir.
class Samsung extends ObservableProduct {

    // Ürün fiyatı değiştikten sonra Observer örneklerine bildirim gönderilir.
    public void changePrice() {
        // Ürün fiyatlarının güncellenmesi vs.

        this.message.productName = "Samsung";
        this.message.message = "Samsung's price updated.";
        this.notifyObserver();
    }
}

File: observer/python/samsung.py
from observable_product import ObservableProduct


class Samsung(ObservableProduct):
    """
    Observer sınıfından türer.
    UML diyagramındaki ConcreteObservable yapısına denk gelmektedir.
    """

    def change_price(self):
        # Ürün fiyatı değiştikten sonra Observer örneklerine bildirim gönderilir.
        # Ürün fiyatlarının güncellenmesi vs.
        self.message.product_name = "Samsung"
        self.message.message = "Samsung's price updated."
        self.notify_observer()
